purchase_request/models/purchase_request.py

```diff
@@ -100,12 +101,28 @@ def _track_subtype(self, init_values):
     is_editable = fields.Boolean(string="Is editable",
                                  compute="_compute_is_editable",
                                  readonly=True)
-
+    to_approve_allowed = fields.Boolean(
+        compute='_compute_to_approve_allowed')
     picking_type_id = fields.Many2one('stock.picking.type',
                                       'Picking Type', required=True,
                                       default=_default_picking_type)
 
     @api.multi
+    @api.depends(
+        'state',
+        'line_ids.product_qty',
+        'line_ids.cancelled',
+    )
+    def _compute_to_approve_allowed(self):
+        for rec in self:
+            rec.to_approve_allowed = (
+                rec.state == 'draft' and
+                sum([
+                    not line.cancelled and line.product_qty or 0
+                    for line in rec.line_ids
+                ]) > 0)
```

Comment:
`rec.state == 'draft' and any(not line.cancelled and line.product_qty for line in rec.line_ids)`
should be faster because `any` stops iteration on the first truthy element.

Reply:
Indeed! I didn't think about using this method. Thanks!